modules/core/src/core/lib/layer.js

```diff
@@ -381,9 +381,44 @@ export default class Layer {
     }
 
     // TODO: Optimize this to use sub-buffer update!
-    const model = this.getSingleModel();
-    if (model) {
-      model.setAttributes({instancePickingColors: attribute});
+    const models = this.getModels();
+    if (models) {
+      models.forEach(model => model.setAttributes({instancePickingColors: attribute}));
+    }
+  }
+
+  // This method overwrittes the current buffer and is used for multi-picking
+  _updateRegularPickingColors(excludeColors) {
+    const {pickingColors} = this.getAttributeManager().attributes;
+    const attribute = pickingColors.state;
+    attribute.update.call(this, attribute);
+    const {value} = attribute;
+
+    // TODO: Optimize this to be faster
+    excludeColors.forEach(color => {
```

Comment:
Do we need to iterate over all excludeColors every time? Can't we just keep incrementally removing the last found color?

Reply:
I wanted to do this but it would behave differently than ```_updateInstancePickingColors```